tests/unit/s2n_async_pkey_test.c

```diff
@@ -300,6 +424,66 @@ int main(int argc, char **argv)
 
     EXPECT_SUCCESS(s2n_cert_chain_and_key_free(chain_and_key));
 
+    struct s2n_hash_state digest = { 0 };
+    EXPECT_SUCCESS(s2n_hash_new(&digest));
+    EXPECT_SUCCESS(s2n_hash_init(&digest, S2N_HASH_SHA1));
+    EXPECT_SUCCESS(s2n_hash_update(&digest, test_digest_data, test_digest_size));
+
+    /* Test: signature offload. */
+    {
+        struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+        EXPECT_NOT_NULL(conn);
+        conn->config->async_pkey_cb = async_pkey_signature_callback;
+
+        EXPECT_FALSE(s2n_result_is_ok(s2n_async_pkey_sign(conn, S2N_SIGNATURE_ECDSA, &digest, s2n_async_sign_complete)));
+        EXPECT_TRUE(s2n_errno == S2N_ERR_ASYNC_BLOCKED);
+
+        EXPECT_SUCCESS(s2n_async_pkey_op_apply(pkey_op, conn));
+        EXPECT_SUCCESS(s2n_async_pkey_op_free(pkey_op));
```

Comment:
All your assertions are in your callbacks. Is there anything you can verify from main to check that they all triggered? Maybe the end effect on the connection?

Reply:
I am not sure, my guess is there is nothing of interest in the connection object, since we only did a signature or decrypt operation, with no negotiation

Reply:
Maybe set a flag from your on_complete handler than? Just a sanity check to make sure your other tests are actually being run.

Reply:
Yes let me add a flag for each callback